packages/fxa-auth-server/lib/routes/utils/oauth.js

```diff
@@ -65,8 +68,10 @@ module.exports = {
     });
 
     // Email the user about their new device connection
-    // (but don't send anything if it was an existing device)
+    // (but don't send anything if it was an existing device or new account)
     if (!credentials.deviceId) {
+      const account = await db.account(credentials.uid);
+      const isNewAccount = Date.now() - account.createdAt < MAX_NEW_ACCOUNT_AGE;
```

Comment:
When creating a new account on a device, the account is already considered verified at this point. Is there a different way to detect that this is a new account and not to send the email?

Reply:
I can't think of a more clever way to do this that doesn't involve a major refactor.

The trouble we're having here is that we first create the account, and then in a second operation we create the requested refresh token. For non-oauth devices we avoid this problem by creating both the account and the sessionToken as a single operation, which means we can ensure we send exactly one email.

I can imagine a future world where there is an atomic "create a new account and provision these oauth tokens" operation that would give us a similar property for oauth clients, but that's a serious refactor!

Reply:
FWIW, I've went with the passing scopes into the session_verify function and then checking that one of the scopes was the "old sync scope"